Mathlib/Topology/LocallyConstant/Algebra.lean

```diff
@@ -291,48 +291,133 @@ theorem coe_algebraMap (r : R) : ⇑(algebraMap R (LocallyConstant X Y) r) = alg
 
 end Algebra
 
+section coeFn
+
+/-- `FunLike.coe` as a `RingHom`. -/
+@[simps!] def coeFnRingHom [Semiring Y] : LocallyConstant X Y →+* X → Y where
+  toMonoidHom := coeFnMonoidHom
+  __ := coeFnAddMonoidHom
+
+/-- `FunLike.coe` as a linear map. -/
+@[simps] def coeFnₗ (R : Type*) [Semiring R] [AddCommMonoid Y]
```

Comment:
```suggestion
@[simps!] def coeFnₗ (R : Type*) [Semiring R] [AddCommMonoid Y]
```

You can see the generated lemmas by adding `?`, and in general it's always better to check them. Here, you see that without the `!` the lemma contains `ZeroHom.toFun`, which is bad because it's not in `simp` normal form. One of the things the `!` does is applying `simp` to the right hand side of the generated lemma to make sure that this doesn't happen. This is a bit slower so it's not the case by default, but you should basically never hesitate to do it if the lemma gets better.